pkg/sanity/controller.go

```diff
@@ -854,6 +855,45 @@ var _ = DescribeSanity("Controller Service", func(sc *SanityContext) {
 			cl.UnregisterVolume(name)
 		})
 
+		It("should fail when publishing volumes more than the node max attach limit", func() {
+
+			By("getting node info")
+			nodeInfo, err := n.NodeGetInfo(
+				context.Background(),
+				&csi.NodeGetInfoRequest{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(nodeInfo).NotTo(BeNil())
+
+			if int(nodeInfo.MaxVolumesPerNode) <= 0 {
+				Skip("No MaxVolumesPerNode")
+			}
+
+			nid := nodeInfo.GetNodeId()
+			Expect(nid).NotTo(BeEmpty())
+
+			// Store the volume name and volume ID for later cleanup.
+			createdVols := map[string]string{}
+			By("creating volumes")
+			for i := 0; i < int(nodeInfo.MaxVolumesPerNode); i++ {
+				name := uniqueString(strings.Join([]string{"sanity-max-attach-limit-vol", strconv.Itoa(i)}, "-"))
```

Comment:
I think `fmt.Sprintf` will be easier to read than this `strings.Join` + `strconv.Itoa`.